server/src/main/java/org/opensearch/transport/TransportService.java

```diff
@@ -1241,6 +1242,40 @@ public <Request extends TransportRequest> void registerRequestHandler(
         transport.registerRequestHandler(reg);
     }
 
+    /**
+     * Registers a new request handler with admission control support
+     *
+     * @param action                The action the request handler is associated with
+     * @param executor              The executor the request handling will be executed on
+     * @param forceExecution        Force execution on the executor queue and never reject it
+     * @param canTripCircuitBreaker Check the request size and raise an exception in case the limit is breached.
+     * @param admissionControlActionType Admission control based on resource usage limits of provided action type
+     * @param requestReader               The request class that will be used to construct new instances for streaming
+     * @param handler               The handler itself that implements the request handling
+     */
+    public <Request extends TransportRequest> void registerRequestHandler(
```

Comment:
How do we ensure plugins espl search or indexing focused start using this interface?

Reply:
Based on last discussion we decided not go with explicit mapping instead each action will register as one of the admissionControlAction type based on which the limits are applied. Otherwise we need explicit action to limits map which is problem for plugins in this way plugins with this interface.

Reply:
For existing plugins , can we create an issue to use this new interface if admission control support is needed ? For new plugins, I think the developer has to make sure they use this interface method. Maybe more documentation to callout the same